kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractDockerMojo.java

```diff
@@ -403,6 +407,16 @@ public RuntimeMode getConfiguredRuntimeMode() {
         return RuntimeMode.KUBERNETES;
     }
 
+    @Override
+    public final void execute() throws MojoExecutionException, MojoFailureException {
+        init();
+        if (!canExecute()) {
+            log.info("`%s` goal is skipped.", mojoExecution.getMojoDescriptor().getFullGoalName());
+            return;
+        }
+        doExecute();
+    }
```

Comment:
I understand that this reorganization is similar to what was discussed in https://github.com/eclipse/jkube/pull/1198#pullrequestreview-847513682

However, there's already an issue to cover this (#147). Code reorganization at this point might make it harder to refactor later on. Especially since this class now has an `execute`, `executeInternal`, and `doExecute` methods with no clear separation of purposes (at least semantically speaking).

For the current PR I would rather not change (complicate) more the structure of the class (although I do agree the execute method is more readable now), end delay that effort until the whole refactor (+ maybe removal) of the class.

Reply:
semantically:
- `execute` is the public method
- `executeInternal` is the execution implementation extended by subclasses -> should be abstract
- I added `doExecute` which is the execution code from the abstract class. executed if not skipped

Basically at the end of the day after refactoring
`execute()`
should call `doExecute`
that should call `executeInternal`.

I am open to a better naming or suggestions.

Reply:
I understand what each does, but there's no clear separation of concerns. If all of the methods are executing stuff, why are you separating two of them... i.e. naming is fine, but separation isn't.

Anyway, this refactor right now introduces more burden if we want to tackle #147, which is the cleaner approach. The point here would be to add the simple log message and try to move on without changing too much what we already have. IMO it would be better to spend the time creating tests that would ensure a successful latter refactor.

Reply:
I initially refactored because it was easier to test.
I refactored again. That should help to tackle  https://github.com/eclipse/jkube/issues/147 IMHO

Reply:
I think I might not have explained myself clear.

If the refactor is required to improve or ease the test scenarios, then it's OK, period.

If the refactor or reorganization is to clean up, improve readability etc. what I was trying to say is that this effort will be wasted if we finally implement 147 (which we should). It might even complicate things further. Efforts should be pushed towards testing the current functionalities of the final Mojos so that the upcoming refactor can be done in a safe way.

Anyway, it's OK as it is. Let's move on.